refactor

kiedtl opened the pull request:
### done
- move each information field into separate files and classes (i.e. `struct`/`impl` pairs, whatever they're called)
- update help messages for arguments
- remove all the `takes_value(false)`
- change all flags to positive
- refactor output

### todo
- fix and test pkgs/mpd (will be completed in another PR)

This PR is finished and ready to be merged.

kiedtl commented:
@Phate6660 when you test the this PR, ensure that you REVERT the latest commit, which breaks the build. I'm going to fix that tomorrow.

kiedtl commented:
This PR is, to the best of my knowledge, fairly complete. MPD (Music) still needs to be tested, however.

Phate6660 commented:
That was on accident, had wrong window selected and my typing did that.

Phate6660 commented:
I'm having weird errors with mpd, so I don't know if I'll be able to test it.

However, the cpu part results in an os error 2.

kiedtl commented:
@Phate6660 this is now ready to review. I've changed the cpu field to not display frequency if it can't find the file.

Other problems (such as the ones on NetBSD, where there isn't any `/proc` will be fixed in the next release.)

Phate6660 commented:
I've already looked through it a few times over the course of the pull request, and think it's good. And since you fixed the os error 2, I think this is ready for merge. I'm gonna test this one more time, and then I'll merge it.

Phate6660 commented:
Ok, so a few things.

CPU outputs: `CPU  =   (2)` (there are borders, I just didn't include them).

The package count is incorrect, at least for `apt`. According to `rsfetch`, I have 91153 packages installed in my Xubuntu VM.

Phate6660 commented on src/music.rs:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    pub fn get(&mut self) -> Result<()> {
+        let data = Command::new("mpc").arg("current")
```
`.arg("current")` is not needed. `current` is the default option for `mpc`.

Phate6660 commented on src/pkgs.rs:
```diff
@@ -0,0 +1,116 @@
+                    .output().context(Pkgcount)?,
+                PkgManager::Alpine  => Command::new("apk").arg("info")
+                    .output().context(Pkgcount)?,
+                PkgManager::Gentoo  => Command::new("qlist").arg("-I")
```
This can be found better as I stated in https://github.com/rsfetch/rsfetch/issues/51#issuecomment-565910597. On top of that, it'll be one less extra executable to be required or called. You don't need to do this right now or anything, just thought I'd bring it to your attention. I can always do it later as well, if you don't want to. I've just been hesitant on touching any of the files while you were still going through and changing/refactoring things.

kiedtl replied:
This can be done in another release, I think. Anyway, I'd prefer to complete this in another PR, since this one was supposed to be solely a refactor, not changing anything in the code that wasn't really necessary.

Phate6660 replied:
Yep, got it. Resolving.

kiedtl commented:
> CPU outputs: CPU = (2) (there are borders, I just didn't include them).

I'd rather fix this in another PR, since this was supposed to be solely a refactor.

> The package count is incorrect, at least for apt. According to rsfetch, I have 91153 packages installed in my Xubuntu VM.

That's because the original code counted the number of total packages in the repositories, not the ones installed.

EDIT: I've fixed this.

> rsfetch seems to be unable to read the various $XDG_* variables. Usually, rsfetch would be able to read them and say that my current DE was XFCE. but this time it just complained about the missing .xinitrc and didn't output anything.

fixed.

Phate6660 commented:
Awesome! Looks like this PR should be ready for merge then, right? I've already looked through the PR numerous times throughout it's commits, and I think it looks good. So you are free to merge it whenever you feel it's ready.

kiedtl commented:
Whew! :)
